src/Workleap.OpenApi.MSBuild/Spectral/AdoCiReportRenderer.cs

```diff
@@ -0,0 +1,13 @@
+namespace Workleap.OpenApi.MSBuild.Spectral;
+
+internal sealed class AdoCiReportRenderer : ICiReportRenderer
+{
+    public async Task AttachReportToBuildAsync(string reportPath)
+    {
+        // Attach the report to the build summary
+        if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_NAME")))
```

Comment:
`AGENT_NAME` might not be specific enough to Azure DevOps. [We detect](https://github.com/gsoft-inc/wl-extensions-xunit/blob/bd7d883917a1915b5af54d4fe848594bc4dab1ca/src/Workleap.Extensions.Xunit/BaseIntegrationFixture.cs#L77-L79) `SYSTEM_TEAMFOUNDATIONCOLLECTIONURI` (Team Foundation = previous name of Azure DevOps, which used to be called VSTS at some point too) in this project and it was based on a library that is specialized in CI detection.